### PR TITLE
Specify that this will use the hosts file and not only DNS

### PR DESCRIPTION
**Pull request by nycdotnet**
This method uses hostname resolution rules like `ping` (uses operating system host resolution rules, at least on Windows), not like `nslookup`/`dig` (only ever queries DNS and ignores local host resolution functionality).

## Summary

The documentation for this method seems to imply that it queries DNS directly (like `nslookup`/`dig` tools do).   I verified that it does not - instead it uses normal host OS resolution rules like `ping` does (I only tested this on Windows).

If you run a console program like this, and edit your `hosts` file to include a line like this `192.168.1.1 my-fun-example.local`, you can keep editing the `hosts` file and see the program's output change every time you save:

```csharp
    class Program
    {
        static async Task Main(string[] args)
        {
            for (int i = 0; i < 1000; i++)
            {
                Thread.Sleep(1000);
                var entry = await Dns.GetHostEntryAsync("my-fun-example.local");

                Console.WriteLine(string.Join("/", entry.AddressList.Select(a => a.ToString()).ToArray()));
                Console.WriteLine(entry.AddressList.Length);
            }
        }
    }
```

### CODE REVIEW

**Comment by davidsh** [comment]
This PR is accurately describing the behavior.  Although this phrase *"This method follows standard hostname resolution rules on Windows"* could be reworded.  It is not so much as "rules" that are followed.  It is that the System.net Dns class does name resolution and is implemented with Win32 APIs like [getaddrinfo](https://docs.microsoft.com/en-us/windows/desktop/api/ws2tcpip/nf-ws2tcpip-getaddrinfo) which uses multiple protocols such as Netbios, LLMNR, DNS, etc.  And for .NET Core on Linux, the Dns class uses Linux equivalents of 'getaddrinfo'.

It is true that the .NET "Dns" class is poorly named.  We've had many requests to enhance the API and make it work like a real DNS client and provide methods to return A, AAAA, CNAME, SRV, TXT records from DNS lookups. But that would require a new class probably named such as "DnsClient", etc.

**Comment by nycdotnet** [comment]
Thank you @davidsh - that was what I was trying to say. Effectively, this class provides a normal hostname resolution experience, and does not act specifically as a pure "DNS client".

**Comment by davidsh** [comment]
>Thank you @davidsh - that was what I was trying to say.

Feel free to re-use my words and post a new commit to this PR.

File: xml/System.Net/Dns.xml [hunk 1]
```diff
@@ -708,7 +708,9 @@
  If an empty string is passed as the `hostNameOrAddress` argument, then this method returns the IPv4 and IPv6 addresses of the local host.
   
  IPv6 addresses are filtered from the results of the <xref:System.Net.Dns.GetHostAddresses%2A> method if the local computer does not have IPv6 installed. As a result, it is possible to get back an empty <xref:System.Net.IPAddress> instance if only IPv6 results were available for the `hostNameOrAddress` parameter.  
-  
+ 
+ This method follows standard hostname resolution rules on Windows, so if a host is described in the `hosts` file, the IP address(es) there will be returned instead of querying DNS.
```
Comment (mairaw) [inline xml/System.Net/Dns.xml]: I know you're probably going to make changes to your PR, but don't (es) to plurals since this is hard to translate.

Source: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/s/s-es

**Comment by nycdotnet** [comment]
@davidsh @mairaw I've revised this - I hope the new version is better.  I sort of wanted to hedge "on a typically configured OS" but thought it was already a bit too wordy and the point is that it's "not actually a DNS client".
